File: v1-sdd-only/client/src/api/wordBook.ts
import request from './request'

export interface WordBookData {
  id: number
  name: string
  description: string
  wordCount: number
  sortOrder: number
  createdAt: string
}

export function getWordBooks() {
  return request.get<{ code: number; message: string; data: WordBookData[] }>('/word-books')
}

File: v1-sdd-only/server/src/main/java/com/ebbinghaus/vocab/domain/entity/WordBook.java
package com.ebbinghaus.vocab.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("word_book")
public class WordBook {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String description;

    private Integer wordCount;

    private Integer sortOrder;

    private LocalDateTime createdAt;
}
